src/components/CodeView/index.spec.tsx

```diff
@@ -48,36 +63,48 @@ describe(__filename, () => {
     );
   };
 
+  type RenderWithLinterProviderParams = Partial<LinterProviderInfo> &
+    RenderParams;
+
+  const renderWithLinterProvider = ({
+    messageMap = undefined,
+    messagesAreLoading = false,
+    selectedMessageMap = undefined,
+    ...renderParams
+  }: RenderWithLinterProviderParams = {}) => {
+    const root = render(renderParams);
+
+    return simulateLinterProvider(root, {
+      messageMap,
+      messagesAreLoading,
+      selectedMessageMap,
+    });
+  };
+
   const renderWithMount = ({
     location = createFakeLocation(),
     ...otherProps
   }: RenderParams = {}) => {
     return mount(
-      <CodeView {...getProps(otherProps)} />,
+      <Provider store={configureStore()}>
+        <CodeView {...getProps(otherProps)} />
+      </Provider>,
```

Comment:
Can we pass `store` in the `context` instead of using `Provider`?

Reply:
whoops, I forgot this code was already passing options to `mount()`. I made that change but I had to add `childContextTypes` which feels wrong. I guess the `<Provider>` approach was nice because it abstracted all of that away. I dunno, I'll just use thte `childContextTypes` approach since it's already in use for the router. 🤷‍♀️

Reply:
Yeah, I think I mentioned this in another issue. We cannot simply remove `PropTypes` because we're using TS. The old context API uses `PropTypes` to pass metadata, which is weird but it is how it is..

Reply:
The invariant exception is a runtime error so it's more than TS. I guess it's React that throws the error but that is confusing. IIRC this is a development mode feature.